Add support for defer script tag

jeroenbell opened the pull request:
Similar to the `async` script tag, this adds an option `js.useDefer` which adds `<script defer ...>`. When enabling both options, most browsers will prefer `async`, and fall back to `defer` when support is limited.

Commit:
Add support for defer script tag

jeroenbell commented:
@sir-dunxalot what do you think of this proposal?

sir-dunxalot commented on index.js:
```diff
@@ -229,10 +230,17 @@ module.exports = {
       }
-      return '<script src="' + path + '"></script>\n';
+  
+      return '<script' + scriptAttributes.join('') + ' src="' + path + '"></script>\n';
```
What do you think about adding the space in the `join()`? That seems less prone to future contribution error. For example:

```js
scriptAttributes.push('async');
```

Then
```js
return '<script' + scriptAttributes.join(' ') + ' src="' + path + '"></script>\n';
```

sir-dunxalot commented:
Hey @jeroenbell, I think this is OK! The only reason I am tentative is because I do not know if there will be any side effects of using `defer` in Ember apps. Have you been able to manually test this? I assume this will be fine but I'll see if I can find any info on using `async` and `defer` with Ember unless you provide a link to some docs, etc.

Left a minor comment on the PR too - your call whether you want to address the comment.

Commit:
Join on spaces

jeroenbell commented:
> Hey @jeroenbell, I think this is OK! The only reason I am tentative is because I do not know if there will be any side effects of using `defer` in Ember apps. Have you been able to manually test this? I assume this will be fine but I'll see if I can find any info on using `async` and `defer` with Ember unless you provide a link to some docs, etc.
> 
> Left a minor comment on the PR too - your call whether you want to address the comment.

I've addressed your comment 🙂 

We have done some internal testing with one of our applications using `async` and `defer` for SEO reasons, and have found no problems so far. But you're right about the fact there isn't much to find about this online.

One thing I would like to add, is when using either `async` or `defer`, it would be best to add the `{{content-for 'js'}}` hook in the `<head>` tag (https://flaviocopes.com/javascript-async-defer/). So it would be best to update the docs as well.

sir-dunxalot commented:
Thanks @jeroenbell. I'm going to do a bit more reading but think this change is fine, especially since this is an 'opt-in' feature of sorts. Documentation and tests will need updating but that can be done after merge, before release.

sir-dunxalot commented:
Released in [2.3.0](https://github.com/sir-dunxalot/ember-cli-concat/releases/tag/2.3.0).
